pyodide_build/build_env.py

```diff
@@ -305,3 +305,19 @@ def local_versions() -> dict[str, str]:
         "pyodide-build": __version__,
         # "emscripten": "TODO"
     }
+
+
+def _create_constraints_file() -> str:
+    try:
+        constraints = get_build_flag("PIP_CONSTRAINT")
+    except ValueError:
+        return ""
+
+    if not constraints:
+        return ""
+
+    constraints_file = Path(constraints)
+    if not constraints_file.is_file():
+        constraints_file.parent.mkdir(parents=True, exist_ok=True)
+        constraints_file.write_text("")
+    return constraints
```

Comment:
```suggestion
def _create_constraints_file() -> str:
    try:
        constraints = get_build_flag("PIP_CONSTRAINT")
    except ValueError:
        return ""

    if not constraints:
        return ""

    constraints_file = Path(constraints).resolve()
    if not constraints_file.is_file():
        constraints_file.parent.mkdir(parents=True, exist_ok=True)
        constraints_file.write_text("")
    return constraints
```

`resolve()`ing the `Path` should also make it track symlinks, and I feel it is generally better, unless there is a special case.

Also, a question: [the `pip` documentation](https://pip.pypa.io/en/latest/cli/pip_install/#cmdoption-c) says that it can be supplied multiple times. Is that also possible through the environment variable?

Reply:
It seems that it is possible, using spaces to delimit multiple entries:
https://github.com/pypa/pip/blob/main/src/pip/_internal/cli/parser.py?plain=1#L248-L250

As a corollary to this, it is NOT possible to specify via the environment variable any path with spaces in it. But hopefully nobody will clone Pyodide into a folder with spaces in the name. Maybe I'll put a check for it.

Reply:
Opened an issue in pip:
https://github.com/pypa/pip/issues/13283